miso-web/src/main/webapp/scripts/sample_ajax.js

```diff
@@ -1243,32 +1274,138 @@ Sample.ui = {
       }
     });
   },
+
+  /**
+   * Returns an array of sample classes associated with selected samples
+   */
+  getSampleClassesForSelected: function () {
+    var selectedIds = Sample.ui.getSelectedIds();
+    
+    // get unique sample class IDs
+    var selectedSampleClasses = jQuery('#listingSamplesTable').dataTable().fnGetData().filter(function (row) {
+      return selectedIds.indexOf(row.id) != -1;
+    }).map(function (dtRow) {
+      return Sample.sampleClasses.filter(function (sc) {
```

Comment:
Shouldn't this return exactly one item? If it returned that, there'd be no need for the `concat`.